lighthouse-core/test/fraggle-rock/scenarios/pptr-test-utils.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+import puppeteer from 'puppeteer';
+import {Server} from '../../../../lighthouse-cli/test/fixtures/static-server.js';
+
+/** @typedef {InstanceType<typeof import('../../../../lighthouse-cli/test/fixtures/static-server.js').Server>} StaticServer */
+
+export function createTestState() {
+  /** @param {string} name @return {any} */
+  const any = name => new Proxy({}, {get: () => {
+    throw new Error(`${name} used without invoking \`state.beforeAll\``);
+  }});
+
+  return {
+    browser: /** @type {import('puppeteer').Browser} */ (any('browser')),
+    page: /** @type {import('puppeteer').Page} */ (any('page')),
+    server: /** @type {StaticServer} */ (any('server')),
+    serverBaseUrl: '',
+
+    async beforeAll() {
+      this.server = new Server();
+      await this.server.listen(0, '127.0.0.1');
+      this.serverBaseUrl = `http://localhost:${this.server.getPort()}`;
+      this.browser = await puppeteer.launch({
+        headless: true,
+      });
+    },
+    async beforeEach() {
+      this.page = await this.browser.newPage();
+    },
+    async afterEach() {
+      await this.page.close();
+    },
+    async afterAll() {
+      await this.browser.close();
+      await this.server.close();
+    },
```

Comment:
Is it possible to run the corresponding jest hooks (`beforeAll`, `beforeEach`, `afterEach`, `afterAll`) in this function? Otherwise I think descriptive names would be helpful like `setupState`, `setupPage`, `cleanupPage`, `cleanupState`.

Reply:
> Is it possible to run the corresponding jest hooks (beforeAll, beforeEach, afterEach, afterAll) in this function?

Yep we can combine all of them into a single invocation.